project/dexter-monitor/routes/user.js

```diff
@@ -204,4 +204,144 @@ exports.getUserCountByProjectName = function(req, res) {
+            return [];
+        });
+}
+
+function updateUserStatusList() {
+    let promises = [];
+    let userStatusList = [];
+    let userList = [];
+
+    promises.push(new Promise((resolve, reject) => {
+        loadUserStatusList()
+            .then((rows) => {
+                userStatusList = rows;
+                resolve();
+            })
+            .catch((err) => {
+                log.error(err);
+                reject();
+            })
+    }));
+    promises.push(new Promise((resolve, reject) => {
+        loadUserList()
+            .then((rows) => {
+                userList = rows;
+                resolve();
+            })
+            .catch((err) => {
+                log.error(err);
+                reject();
+            })
+    }));
+
+    return Promise.all(promises)
+        .then(() => {
+            return Promise.map(userStatusList, (row) => {
+                if (_.findIndex(userList, (user) => user.userId == row.userId) >= 0) {
+                    row.dexterYn = 'Y';
+                } else if (row.dexterYn == 'Y') {
+                    row.dexterYn = 'N';
+                } else {
+                    return;
+                }
+
+                const sql =
+                    `UPDATE DexterUserList
+                     SET dexterYn='${row.dexterYn}', reason='', ide='', language=''
+                     WHERE userId='${row.userId}'`;
+                return database.exec(sql)
+                    .catch((err) => {
+                        log.error('Failed to set dexterYn : ' + err);
+                    });
+            });
+        });
+}
+
+function loadAndCreateUserStatusTable() {
+    const sql = `SELECT DexterUserList.userLab AS groupName,
+                        COUNT(DexterUserList.userId) AS allDeveloperCount,
+                        installedDeveloperCount,
+                        nonTargetDeveloperCount
+                 FROM DexterUserList
+                 LEFT JOIN (SELECT userLab, COUNT(DUL1.userId) AS installedDeveloperCount
+                            FROM DexterUserList AS DUL1
+                            WHERE DUL1.dexterYn='y'
+                            GROUP BY DUL1.userLab) AS d1
+                 ON DexterUserList.userLab = d1.userLab
+                 LEFT JOIN (SELECT userLab, COUNT(DUL2.userId) AS nonTargetDeveloperCount
+                            FROM DexterUserList AS DUL2
+                            WHERE DUL2.dexterYn='u'
+                            GROUP BY DUL2.userLab) AS d2
+                 ON DexterUserList.userLab = d2.userLab
+                 GROUP BY DexterUserList.userLab`;
+    return database.exec(sql)
+        .then((rows) => {
+            rows.forEach((row) => {
+                row.targetDeveloperCount = row.allDeveloperCount - row.nonTargetDeveloperCount;
+                row.installationRate = (row.installedDeveloperCount / row.targetDeveloperCount * 100).toFixed(2);
+            });
```

Comment:
Please separated by a function.
When we have to change calculation for ratio, We have to check only this function.
getTargetDeveloperCount()/ getInstallationRatio()